### PR TITLE
feat: add dyanmic compute unit fee calculation

### PR DESCRIPTION
**Pull request by ali-bahjati**

### CODE REVIEW

File: config/config.toml [hunk 1]
```diff
@@ -79,6 +79,10 @@ key_store.mapping_key = "RelevantOracleMappingAddress"
 # Price per compute unit offered for update_price transactions
 # exporter.compute_unit_price_micro_lamports =
 
+# Weather the dynamic compute unit pricing is enabled. When enabled, the compute unit price is
```
Comment (guibescos) [inline config/config.toml:82]: whether

File: src/agent/solana/exporter.rs [hunk 2]
```diff
@@ -321,41 +429,15 @@ impl Exporter {
...
-            //   handle_key_requests() in remote_keypair_loader.rs
-
-            debug!(
-                self.logger,
-                "Exporter: Publish keypair is None, requesting remote loaded key"
-            );
-            let kp = RemoteKeypairLoader::request_keypair(&self.keypair_request_tx).await?;
-            debug!(self.logger, "Exporter: Keypair received");
-            kp
-        };
```
Comment (guibescos) [inline src/agent/solana/exporter.rs:358]: Why did this get deleted

Comment (ali-bahjati) [reply src/agent/solana/exporter.rs:358]: This is not deleted. I moved it to another function so I can reuse it.

File: src/agent/solana/exporter.rs [hunk 3]
```diff
@@ -261,45 +280,134 @@ impl Exporter {
...
-    ///
-    /// This design is intended to:
-    /// - Decouple the rate at which the local store is updated and the rate at which we publish transactions.
-    ///   A user sending an unusually high rate of price updates shouldn't be reflected in the transaction rate.
-    /// - Degrade gracefully if the blockchain RPC node exhibits poor performance. If the RPC node takes a long
-    ///   time to respond, no internal queues grow unboundedly. At any single point in time there are at most
-    ///   (n / batch_size) requests in flight.
-    async fn publish_updates(&mut self) -> Result<()> {
+    async fn update_recent_compute_unit_price(&mut self) -> Result<()> {
+        let permissioned_updates = self.get_premissioned_updates().await?;
```
Comment (guibescos) [inline src/agent/solana/exporter.rs:314]: get_premissioned_updates -> get_permissioned_updates

File: src/agent/solana/exporter.rs [hunk 4]
```diff
@@ -261,45 +280,134 @@ impl Exporter {
...
+            .into_iter()
+            .collect::<Result<Vec<_>, _>>()
+            .context("Failed to get recent prioritization fees")?;
+
+        prioritization_fees_batches
+            .iter()
+            .for_each(|prioritization_fees| {
+                prioritization_fees.iter().for_each(|fee| {
+                    // Get the maximum prioritaztion fee over all fees retrieved for this slot
+                    let prioritiztion_fee = slot_compute_fee
```
Comment (guibescos) [inline src/agent/solana/exporter.rs:347]: prioritization

**Pull request merged**